onnxruntime/core/common/cpuid_info.cc

```diff
@@ -103,36 +645,142 @@ CPUIDInfo::CPUIDInfo() {
 #ifdef HWCAP_ASIMDDP
 
     // only works on ARM linux or android, does not work on Windows
+    // Assuming no hyper-threading, no NUMA groups
     if (pytorch_cpuinfo_init_) {
       is_hybrid_ = cpuinfo_get_uarchs_count() > 1;
       has_arm_neon_dot_ = cpuinfo_has_arm_neon_dot();
+      const uint32_t core_cnt = cpuinfo_get_cores_count();
+      core_uarchs_.resize(core_cnt, cpuinfo_uarch_unknown);
+      is_armv8_narrow_ld_.resize(core_cnt, false);
+      for (uint32_t c = 0; c < core_cnt; c++) {
+        const struct cpuinfo_processor* proc = cpuinfo_get_processor(c); 
+        if (proc == nullptr) {
+          continue;
+        }
+        const struct cpuinfo_core* corep = proc->core;
+        if (corep == nullptr) {
+          continue;
+        }
+        auto coreid = proc->linux_id;
+        auto uarch = corep->uarch;
+        core_uarchs_[coreid] = uarch;
+        if (uarch == cpuinfo_uarch_cortex_a53 || uarch == cpuinfo_uarch_cortex_a55r0 ||
+            uarch == cpuinfo_uarch_cortex_a55) {
+          is_armv8_narrow_ld_[coreid] = true;
+        }
+      }
     } else {
       has_arm_neon_dot_ = ((getauxval(AT_HWCAP) & HWCAP_ASIMDDP) != 0);
     }
 
 #elif defined(_WIN32)
-    // TODO implement hardware feature detection in windows.
-    is_hybrid_ = true;
-#endif
-#endif
+
+    // Read MIDR from windows registry
+    // TODO!! Don't support multiple processor group yet!!
+    constexpr int MAX_CORES = 64;
+    TCHAR midrKey[MAX_VALUE_NAME] = "";  // buffer for processor registry name
+    uint32_t lastUarch = cpuinfo_uarch_unknown;
+    for (int i = 0; i < MAX_CORES - 1; i++) {
+      snprintf(midrKey, MAX_VALUE_NAME, "HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\%d", i);
+      uint64_t midrVal;
+      unsigned long midrSize = sizeof(uint64_t);
+
+      // ARM lists for each coprocessor register 5 fields: op0/op1/CRn/CRm/op2. You need to put those numbers through the ARM64_SYSREG macro:
+      //  #define ARM64_SYSREG(op0, op1, crn, crm, op2) \
+      //    (((op0 & 1) << 14) |                        \
+      //     ((op1 & 7) << 11) |                        \
+      //     ((crn & 15) << 7) |                        \
+      //     ((crm & 15) << 3) |                        \
+      //     ((op2 & 7) << 0))
+      //
+      // For the CP value of MIDR, op0 = 3 and the others are all = 0, so we come up with 0x4000,
+      auto retCode = ::RegGetValue(HKEY_LOCAL_MACHINE, midrKey, "CP 4000", RRF_RT_REG_QWORD, nullptr, &midrVal, &midrSize);
```

Comment:
So, this file will only be use in our python build, not the others. Because this function is desktop-only.

Reply:
What should I use then? any pointers? It works on my surface pro X C++ build though.

Reply:
Build our code with "--enable_wcos" , then open vc++ command line tool, run 
```
dumpbin /imports onnxruntime.dll
```
Without your change, kernel32.dll shouldn't be in the output.

Reply:
I'm afraid I can't help you find a replacement. But, it doesn't block you merging this PR. Anyone who wants this feature  still can build from source.  Many of our customers do so.

Reply:
Windows Team said RegGetValue is supported by OneCore, so we should be ok here.

Reply:
We still need them to confirm it is in Windows 8 OneCore. If you are adding a this feature to Windows source code tree, you don't need to have this concern. Because nobody would backport this code to Windows 8. But we are not.